apps/aecore/lib/aecore/sync/chain.ex

```diff
@@ -127,4 +124,39 @@ defmodule Aecore.Sync.Chain do
 
     header_hash
   end
+
+  @doc """
+  Merge two chains in descending order
+  """
+  @spec merge_descending(list(chain()), list(chain())) :: list(chain())
+  def merge_descending(list1, list2) do
```

Comment:
Change to a more semantic correct name, for instance: merge_descending_chains. This function does not work for general lists and is only used in this module, maybe this should be a private function? You can generalize this operation and add it to the ListUtils module by adding an additional argument which decodes the items in the list. For instance:
def merge_descending(list1, list2, getter_fun \\ &(&1)) do
merge(list1, list2, [], getter_fun)
end
...
  defp merge(
         [hd1 | rest1] = list1,
         [hd2 | rest2] = list2,
         acc, getter_fun
       ) do
    height1 = getter_fun(hd1)
    height2 = getter_fun(hd2)
    ...

Then instead of merge_descending(chain_1, chain_2) you could write:
ListUtils.merge_descending(chain_1, chain_2, fn %{height: height} -> height end)

Reply:
I don't see why I'd like this to be generic, when it is used only in the Chain module, will make it private though.